deploy/determined_deploy/aws/cli.py

```diff
@@ -137,7 +138,7 @@ def deploy_aws(args: argparse.Namespace) -> None:
         constants.deployment_types.SIMPLE: simple.Simple,
         constants.deployment_types.SECURE: secure.Secure,
         constants.deployment_types.VPC: vpc.VPC,
-    }
+    }  # type: Dict[str, Union[Type[simple.Simple], Type[secure.Secure], Type[vpc.VPC]]]
```

Comment:
non-blocking: Fine if you want to call this out of scope, but if you change the `DeterminedDeployment` constructor to have the same signature as that of the subclasses (by moving the call to `resource_filename` inside it), you should be able to make this just `Dict[str, Type[base.DeterminedDeployment]]` while also reducing duplication across the classes.

Reply:
I had considered the `__init__` change you mentioned.  I think I'll do it.

But it happens that without the Union, you get an error:

```
determined_deploy/aws/cli.py:156: error: Cannot instantiate abstract class 'DeterminedDeployment' with abstract attribute 'deploy'
```

I could remove the `@abc.abstractmethod` to get rid of that, but I think I prefer the static check to the runtime check.

Reply:
Interesting; I wonder if there's a way to get Mypy to do the thing we want. Anyway, this is fine, given that.